pkg/repart/disk_repart.go

```diff
@@ -221,20 +224,26 @@ func runSystemdRepart(s *sys.System, target string, parts []Partition, flags ...
 		return fmt.Errorf("failed partitioning disk '%s' with systemd-repart: %w", target, err)
 	}
 	uuids := []struct {
-		UUID    string `json:"uuid,omitempty"`
-		PartNum uint   `json:"partno,omitempty"`
+		UUID string `json:"uuid,omitempty"`
+		File string `json:"file,omitempty"`
 	}{}
 
 	err = json.Unmarshal(out, &uuids)
 	if err != nil {
 		return fmt.Errorf("failed parsing systemd-repart JSON output: %w", err)
 	}
-	if len(uuids) != len(parts) {
-		return fmt.Errorf("partitions mismatch between Deployment and systemd-repart JSON output: %s", string(out))
```

Comment:
I'd suggest to always log the output on a debug level above. It proves to be really important to understand the sequence.